ArduPlane/RC_Channel.cpp

```diff
@@ -78,6 +78,23 @@ void RC_Channel_Plane::do_aux_function_q_assist_state(AuxSwitchPos ch_flag)
     }
 }
 
+void RC_Channel_Plane::do_aux_function_soaring_3pos(AuxSwitchPos ch_flag)
+{
+#if SOARING_ENABLED == ENABLED
+    switch (ch_flag) {
+    case AuxSwitchPos::LOW:
+        plane.g2.soaring_controller.set_enable_ch_state(SoaringController::ActiveStatus::SOARING_DISABLED);
```

Comment:
Perhaps you could change the type of the cached channel state variable?  That would allow for direct assignment.

Putting better names on what those states mean could then be done as an accessor - `SoaringController::ActiveStatus pilot_desired_state() const` or similar.

Reply:
yeah, I'd agree that the ```_ch``` terminology needs to be removed

Reply:
The names are now better. I haven't changed the type - it needs to be mapped from `AuxSwitchPos` to `ActiveStatus` at some point so I'm not sure it would be an overall saving